src/mlpack/tests/serialization.hpp

```diff
@@ -220,9 +220,9 @@ void SerializeObject(T& t, T& newT)
 
     try
     {
-      i >> BOOST_SERIALIZATION_NVP(newT);
+      i >> CEREAL_NVP(newT);
```

Comment:
I believe the way that these functions work is that the "name" in the NVP (name-value pair) is computed from the variable itself.

So, here you would actually want to do `CEREAL_NVP(t)` to match the name from before.  However, the function takes in two arguments `t` and `newT` and they should stay separate... so you might want to do something like this:

For serialization:
```
    try
    {
      T& x(t); // Create a reference to t.
      o << CEREAL_NVP(x);
    }
```

Then for deserialization:

```
    try
    {
      T& x(newT);
      i >> CEREAL_NVP(x);
    }
```

This way the name matches.  Anyway, it seems likely that you might have to do this for some of the other serialization test functions too. :+1: